### PR TITLE
Generalize idinxpres. Add idfnv. Various comment edits. Minor mathbox edits.

### PR DESCRIPTION
**Pull request by benjub**
As for the generalization of ~idinxpres (and the relabeling of the previous idinxpres to idinxpresid and its derivation from idinxpres), I hope credits are ok, in particular for @mazsa.

As for ~idfnv, its proof is rather trivial, but the very short and variable-free statement is appealing enough in my opinion

### CODE REVIEW

File: set.mm [hunk 1]
```diff
@@ -54049,20 +54049,33 @@ the restriction (of the relation) to the singleton containing this
...
   $}
 
+  ${
+    $d x y A $.  $d x y B $.
+    $( The intersection of the identity relation with a cartesian product is
+       the restriction of the identity relation to the intersection of the
+       factors.  (Contributed by FL, 2-Aug-2009.)  (Proof shortened by Peter
+       Mazsa, 9-Sep-2022.)  Generalize statement from cartesian square (now
+       ~ idinxpresid ) to cartesian product.  (Revised by BJ, 23-Sep-2023.) $)
+    idinxpres $p |- ( _I i^i ( A X. B ) ) = ( _I |` ( A i^i B ) ) $=
```
Comment (mazsa) [inline set.mm:54059]: Nice. I have a similar result
```metamath
  ${
    $d A x $.  $d B x $.  $d R x $.
    $( Two ways to say that an intersection of the identity relation with a
       Cartesian product is a subclass.  (Contributed by Peter Mazsa,
       12-Dec-2023.) $)
    ref4 $p |- ( ( _I i^i ( A X. B ) ) C_ R <-> ( _I |` ( A i^i B ) ) C_ R ) $=
      ( vx cid cxp cin wss cv wbr wral cres ref5 idref bitr4i ) EABFGCH
    DIZPCJDABGZKEQLCHDABCMDQCNO $.
  $}
```
based on
```metamath
  ${
    $d A x y $.  $d B x y $.  $d R x y $.
    $( Two ways to say that an intersection of the identity relation with a
       Cartesian product is a subclass.  (Contributed by Peter Mazsa,
       12-Dec-2023.) $)
    ref5 $p |- ( ( _I i^i ( A X. B ) ) C_ R <-> A. x e. ( A i^i B ) x R x ) $=
      ( vy cv wceq wbr wi wral wcel cid cxp cin wss equcom imbi1i breq2
    ralbii ceqsralv2 bitr3i idinxpss ralin 3bitr4i ) AFZEFZGZUEUFDHZI
    ZECJZABJUECKUEUEDHZIZABJLBCMNDOUKABCNJUJULABUJUFUEGZUHIZECJULUNUI
    ECUMUGUHEAPQSUHUKEUECUFUEUEDRTUASAEBCDUBUKABCUCUD $.
  $}
```
but this is more general.

File: set.mm [hunk 2]
```diff
@@ -54049,20 +54049,33 @@ the restriction (of the relation) to the singleton containing this
...
-    $( The intersection of the identity function with a square cross product.
+    $( The intersection of the identity relation with the cartesian square of a
+       class is the restriction of the identity relation to that class.
        (Contributed by FL, 2-Aug-2009.)  (Proof shortened by Peter Mazsa,
        9-Sep-2022.) $)
-    idinxpres $p |- ( _I i^i ( A X. A ) ) = ( _I |` A ) $=
-      ( vy vx cid cxp cin cres wcel cop wceq wrex elidinxpid elrid bitr4i eqriv
-      cv ) BDAAEFZDAGZBPZQHSCPZTIJCAKSRHCASLCSAMNO $.
+    idinxpresid $p |- ( _I i^i ( A X. A ) ) = ( _I |` A ) $=
+      ( cid cxp cin cres idinxpres inidm reseq2i eqtri ) BAACDBAADZEBAEAAFJABAG
```
Comment (mazsa) [inline set.mm:54071]: I think you shortened it now.

**Review by mazsa — APPROVED** [review]

File: set.mm (hunk 1 above)
Comment (benjub) [reply set.mm:54059]: Thanks. Yes, I think idinxpres subsumes ref4 (which can be proved from it using sseq1i).

File: set.mm [hunk 3]
```diff
@@ -534224,22 +534252,14 @@ Identity relation (complements)
...
-
 
 $(
 -.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-
-  Diagonal in a Cartesian square
+  Functionalized identity relation (diagonal in a Cartesian square)
 -.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-
 
-  Definition of the diagonal in the Cartesian square of a set.
+  This subsection defines a functionalized  version of the identity relation,
```
Comment (icecream17) [inline set.mm:534261]:
```suggestion
  This subsection defines a functionalized version of the identity relation,
```

[mathbox]

**Review by icecream17 — APPROVED** [review]

File: set.mm [hunk 4]
```diff
@@ -11820,11 +11821,11 @@ This definition (in the form of ~ dfifp2 ) appears in Section II.24 of
...
     ( wn wo biorf imbi1d ) ADBABECABFG $.
 
-  $( A characterization of when an expression involving nor associates.  This
-     is identical to the case when alternative denial is associative, see
-     ~ nanass .  Remark:  Like alternative denial, nor is also commutative, see
-     ~ norcom .  (Contributed by RP, 29-Oct-2023.)  (Proof shortened by Wolf
-     Lammen, 17-Dec-2023.) $)
+  $( A characterization of when an expression involving joint denials
+     associates.  This is identical to the case when alternative denial is
+     associative, see ~ nanass .  Remark:  Like alternative denial, nor is also
```
Comment (avekens) [inline set.mm:11826]: Then "nor" should be replaced here, too (by "joint denial").

Maybe "(joint denial)" and "(alternative denial)" should be added to the section headers "Logical "nor"" and "Logical "nand" (Sheffer strole)", and "(logical "nor")" and "(logical "nand") to the section headers "Joint denial" and "Alternative denial".

File: set.mm (hunk 3 above)
Comment (benjub) [reply set.mm:534261]: I cannot apply the suggestion on this webpage. If anyone can, please do. I've already done it on my local branch but there are several more changes and I'd like to do that in an upcoming PR.

File: set.mm (hunk 4 above)
Comment (benjub) [reply set.mm:11826]: Thanks, I'll do it in my upcoming PR. I modified it to be consistent with nanass. As for the other places, maybe we should indeed add them, to have both.

"NAND" and "NOR" are like nicknames for "alternative denial" and "joint denial" respectively, but they are widely used, so I don't know if we should prefer one or the other (in titles and their definition, we should provide both). In intuitionistic logic, NOR is indeed "joint denial", but NAND is weaker than "alternative denial". More faithful names would be non-conjunction and non-disjunction.

**Review by avekens — APPROVED** [review]
OK, if my raimaining remarks are respected in an upcoming PR.

**Comment by benjub** [comment]
Actually, I pushed the two requested changes in this PR, since it's cleaner.

**Pull request merged**